src/main/java/nextstep/subway/domain/Line.java

```diff
@@ -64,15 +66,69 @@ public List<Station> getStations() {
         if (this.getSections().isEmpty()) {
             return Collections.emptyList();
         }
-        List<Station> stations = this.getSections().stream()
-                .map(Section::getDownStation)
-                .collect(Collectors.toList());
-        stations.add(0, this.getSections().get(0).getUpStation());
+
+        List<Station> stations = new ArrayList<>();
+        Station nowStation = startStation;
+        stations.add(nowStation);
+        while (!nowStation.equals(endStation)) {
+            for (Section section : sections) {
+                if (section.getUpStation().equals(nowStation)) {
+                    nowStation = section.getDownStation();
+                    stations.add(nowStation);
+                    break;
+                }
+            }
+        }
         return stations;
     }
 
+    public void addSections(Section section) {
+        if (new HashSet<>(this.getStations()).containsAll(List.of(section.getUpStation(), section.getDownStation()))) {
+            throw new BusinessException();
+        }
+
+        if (this.sections.isEmpty()) {
+            this.startStation = section.getUpStation();
+            this.endStation = section.getDownStation();
+            sections.add(section);
+            return;
+        }
+
+        if (this.endStation.equals(section.getUpStation())) {
+            this.endStation = section.getDownStation();
+            sections.add(section);
+            return;
+        }
+        if (this.startStation.equals(section.getDownStation())) {
+            this.startStation = section.getUpStation();
+            sections.add(section);
+            return;
+        }
+
+        Optional<Section> beforeSection = sections.stream()
+                .filter(sectionIter ->
+                        sectionIter.getUpStation().equals(section.getUpStation()) || sectionIter.getDownStation().equals(section.getDownStation()
+                        )).findFirst();
+        beforeSection.orElseThrow(BusinessException::new).splitSection(section);
+        sections.add(section);
+    }
+
+    public Station getStartStation() {
+        return this.startStation;
+    }
+
+    public Station getEndStation() {
+        return this.endStation;
+    }
+
     public void removeSection(Station station) {
-        this.sections.remove(this.sections.size() - 1);
+        for (Section section : sections) {
+            if (section.getDownStation().equals(station)) {
+                endStation = section.getUpStation();
+                this.sections.remove(section);
+                break;
+            }
+        }
```

Comment:
```suggestion
sections.stream()
    .filter(section -> section.getDownStation().equals(station))
    .findFirst()
    .ifPresent(section -> {
        endStation = section.getUpStation();
        sections.remove(section);
    });
```
사람에 따라 다르게 느끼겠지만 스트림을 활용해보아도 좋을 것 같습니다! (예시 코드가 틀릴 수 있음 주의😅)